### DISCUSSION TITLE
Discussions

### DISCUSSION THREAD

**Discussion by Richard-Weiss** [Richard-Weiss/Bing-Creator-Image-Downloader#2] [General]
## Welcome
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

**Comment by jinxyz1** [comment]
Sorry if this is a silly problem. I just tried this today with everything at the most recent version. After installing everything, I get this response when I try to run python .\main.py . Is this an installation error or am I just using it wrong? If you could, any advice would be greatly appreciated as the tool seems absolutely wonderful. I put this here because I assumed I was the issue and not something with the code.

2023-11-18 01:53:19,842 INFO Fetching metadata of collections...
Traceback (most recent call last):
  File "C:\PATH\main.py", line 381, in <module>
    asyncio.run(main())
  File "C:\PATH\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\PATH\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\PATH\main.py", line 369, in main
    await set_creation_dates(image_data)
  File "C:\PATH\main.py", line 254, in set_creation_dates
    await asyncio.gather(*tasks)
  File "C:\PATH\main.py", line 258, in _set_creation_date
    extracted_ids = await _extract_set_and_image_id(image['image_page_url'])
  File "C:\PATH\main.py", line 290, in _extract_set_and_image_id
    image_set_id = result.group('image_set_id')
AttributeError: 'NoneType' object has no attribute 'group'

**Reply by Richard-Weiss** [reply]
Thanks for the heads up.  
It looks like the url it received doesn't match the pattern.  
I'll create an issue from this and a branch for debugging.  
I'll let you know once it's done, so you can fetch the branch and show me the output.

**Comment by OHNOES4** [comment]
I happen to get stuck around step 8 "Run pip install -r .\requirements.txt to install all dependencies (You may need to add the PythonXX\Scripts folder to your PATH first)".  Putting "pip install -r .\requirements.txt" into PowerShell seems to give me the message "ERROR: Could not open requirements file: [Errno 2] No such file or directory: '.\\requirements.txt'". What should i do?

**Reply by Richard-Weiss** [reply]
Are you sure you're in the right directory?

**Reply by OHNOES4** [reply]
Not really, would it happen to be the "Bing-Creator-Image-Downloader-main"  folder?

**Reply by Richard-Weiss** [reply]
Yes, that's the folder of the repository. In there is the requirements.txt file you are referencing, so you will have to change the directory of powershell first. Maybe you can also ask Bing for more help. It's rather late for me.

**Reply by OHNOES4** [reply]
Alright, I'll try it out. Thanks for the help!

**Comment by Semitor** [comment]
Hi,
 first of all I want to say I have 0 experience with any form of coding, so all of this is new to me. I watched some vids to download git, python, and I think I cloned your project correctly, I did set the PATHs, but unfortunetly I get stuck at "Run pip install -r .\requirements.txt to install all dependencies (You may need to add the PythonXX\Scripts folder to your PATH first)", I get:

 ERROR: Could not build wheels for multidict, which is required to install pyproject.toml-based projects. 

I am also not sure if choosing the one of 2 methods before doing that is important or not, I tried the clipboard one but honestly I really don't know what to change in that file. Any ideas where I probably failed and what should I do?

**Reply by Richard-Weiss** [reply]
@Semitor Can you share the whole stacktrace of the pip error?
I only found [this](https://stackoverflow.com/a/77806449) on Stackoverflow but I'm not sure if that's your issue.
In the config.toml is an option called `method` and a description for it.
If you have sub 1000 images you should use the api method, else the file one.

**Reply by Semitor** [reply]
Yes it seems it was the C++, it successfuly installed it but I do get those pop ups: 

 WARNING: The script tabulate.exe is installed in 'C:\Users\Bartek\AppData\Roaming\Python\Python312\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script dotenv.exe is installed in 'C:\Users\Bartek\AppData\Roaming\Python\Python312\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.

So I guess I did fail somewhere during instalation or cloning, but the pip install -r .\requirements.txt worked. Now my question is assuming it will run, how does it work? I clicked the main python.exe in the bing-creator-image-downloader and it showed some lines and shutdown immedieately, the folder you mentioned those images will download to will create itself somewhere?

**Reply by Richard-Weiss** [reply]
@Semitor You have to copy the cookie like described in the readme and set the `method` in the config to `api`. I think the README should explain it well enough for you. So you should do the collection api steps and then skip the file method steps by continuing with the shared next steps.

**Reply by Semitor** [reply]
@Richard-Weiss I got the cookie, set the method to api and skipped file steps but unfortunetly nothing happens so I can't really do much here, Thank you for help, I might reinstall everything and give it another go.

**Comment by nguyenly3009** [comment]
Hi Bro,

I try run main.py but error:

  File "C:\AI\Bing-Creator-Image-Downloader\main.py", line 8, in <module>
    from tomllib import load
ModuleNotFoundError: No module named 'tomllib'

Please help me !

**Reply by nguyenly3009** [reply]
i try it again and successfull. Thank Bro !
